.github/workflows/verify_build.yaml

```diff
@@ -21,3 +21,12 @@ jobs:
           pull: true
           push: false
           tags: test-build
+
+      - name: rust-actions-runner-1.58
+        uses: docker/build-push-action@v2
+        with:
+          context: ./rust-actions-runner/
+          build-args: RUST_VERSION=1.58
+          pull: true
+          push: false
```

Comment:
The whole `build-test-image` is used to check that the changes to the Dockerfile[s] are valid, as it does not push anywhere. It might be poorly named for that, we can remove it now.

I'm not sure I understand the purpose of this new build, if we don't push it: its Dockerfile has a different base that looks like something we would want to use for the GH Agents, so I think it should be actually published.